src/main/java/jcifs/http/NetworkExplorer.java

```diff
@@ -0,0 +1,622 @@
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.bouncycastle.util.encoders.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import jcifs.Address;
+import jcifs.CIFSContext;
+import jcifs.CIFSException;
+import jcifs.Config;
+import jcifs.DfsReferralData;
+import jcifs.NameServiceClient;
+import jcifs.SmbConstants;
+import jcifs.config.PropertyConfiguration;
+import jcifs.context.BaseContext;
+import jcifs.netbios.NbtAddress;
+import jcifs.smb.DfsReferral;
+import jcifs.smb.NtStatus;
+import jcifs.smb.NtlmPasswordAuthentication;
+import jcifs.smb.SmbAuthException;
+import jcifs.smb.SmbException;
+import jcifs.smb.SmbFile;
+import jcifs.smb.SmbFileInputStream;
+
+
+/**
+ * This servlet may be used to "browse" the entire hierarchy of resources
+ * on an SMB network like one might with Network Neighborhood or Windows
+ * Explorer. The users credentials with be negotiated using NTLM SSP if
+ * the client is Microsoft Internet Explorer.
+ * 
+ * @deprecated Unsupported
+ */
+@Deprecated
+public class NetworkExplorer extends HttpServlet {
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = -3847521461674504364L;
+
+    private static final Logger log = LoggerFactory.getLogger(NetworkExplorer.class);
+
+    private String style;
+    private boolean credentialsSupplied;
+    private boolean enableBasic;
+    private boolean insecureBasic;
+    private String realm, defaultDomain;
+
+    private CIFSContext transportContext;
+
+
+    @Override
+    public void init () throws ServletException {
+
+        StringBuffer sb = new StringBuffer();
+        byte[] buf = new byte[1024];
+        int n;
+        String name;
+
+        Properties p = new Properties();
+        p.putAll(System.getProperties());
+        p.setProperty("jcifs.smb.client.soTimeout", "600000");
+        p.setProperty("jcifs.smb.client.attrExpirationPeriod", "300000");
+
+        Enumeration<String> e = getInitParameterNames();
+        while ( e.hasMoreElements() ) {
+            name = e.nextElement();
+            if ( name.startsWith("jcifs.") ) {
+                p.setProperty(name, getInitParameter(name));
+            }
+        }
+
+        try {
+            if ( p.getProperty("jcifs.smb.client.username") == null ) {
+                new NtlmSsp();
+            }
+            else {
+                this.credentialsSupplied = true;
+            }
+
+            try {
+                try ( InputStream is = getClass().getClassLoader().getResourceAsStream("jcifs/http/ne.css"); ) {
+                    while ( ( n = is.read(buf) ) != -1 ) {
+                        sb.append(new String(buf, 0, n, "ISO8859_1"));
+                    }
+                    this.style = sb.toString();
+                }
+            }
+            catch ( IOException ioe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)